pkgs/by-name/cl/clash-verge-rev/package.nix

```diff
@@ -1,19 +1,229 @@
+  };
+  unset_dns = fetchurl {
+    url = "https://github.com/clash-verge-rev/set-dns-script/releases/download/script/unset_dns.sh";
+    hash = "sha256-Nj0bYqUvtxxMVi1Kv5OydWrmO+fhJyMUkSeO8AnjWmo=";
+  };
+  country = fetchurl {
+    url = "https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/country.mmdb";
+    hash = "sha256-eb5jeWzpP/nblMBGTMDsXSYh/rD5jxk/RWpDNFtyhTs=";
+  };
+
+  # overrideAttrs not play well with buildGoModule. Simply redefine a drv.
+  mihomo-alpha = buildGoModule rec {
+    pname = "mihomo";
+    version = "1.18.6-unstable-2024-06-28";
+
+    src = fetchFromGitHub {
+      owner = "MetaCubeX";
+      repo = "mihomo";
+      rev = "0e228765fce4d709af1e672426dea5294e6b7544";
+      hash = "sha256-ZrNnFkkvS8hLEW6u9ZOZ9icehkqlpI4iA3lQf7wM0tg=";
+    };
+
+    vendorHash = "sha256-lBHL4vD+0JDOlc6SWFsj0cerE/ypImoh8UFbL736SmA=";
+
+    excludedPackages = [ "./test" ];
+
+    ldflags = [
+      "-s"
+      "-w"
+      "-X github.com/metacubex/mihomo/constant.Version=${version}"
+    ];
+
+    tags = [ "with_gvisor" ];
+
+    # network required
+    doCheck = false;
+  };
+in
+rustPlatform.buildRustPackage rec {
   pname = "clash-verge-rev";
-  version = "1.6.6";
+  version = "1.7.2-unstable-2024-07-04";
 
-  src = fetchurl {
-    url = "https://github.com/clash-verge-rev/clash-verge-rev/releases/download/v${version}/clash-verge_${version}_amd64.deb";
-    hash = "sha256-x+Xkasg6Yzft5CLg2YFCRkgpDeiVvvdmcLjrg+oIOT8=";
+  src = fetchFromGitHub {
+    owner = "clash-verge-rev";
+    repo = "clash-verge-rev";
+    rev = "5b7b3be6f9623ccb04bf96972f08f6146cf866dd";
+    hash = "sha256-3RyqvYquuX52o+TfD+NnwlQ/laRoquwhVnMOEEz9eWU=";
   };
 
-  meta = old.meta // (with lib; {
+  ui = stdenvNoCC.mkDerivation {
+    inherit src version;
+    pname = "${pname}-ui";
+
+    pnpmDeps = pnpm.fetchDeps {
+      inherit pname version src;
+      hash = "sha256-PEK07nXWoemhqYAHp2+feh7OpMlMDdp+OyJLh52F7S4=";
+    };
+
+    nativeBuildInputs = [
+      nodejs
+      pnpm.configHook
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      pnpm web:build
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      cp -r dist $out
+
+      runHook postInstall
+    '';
+  };
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoLock = {
+    lockFile = "${src}/src-tauri/Cargo.lock";
+    outputHashes = {
+      "sysproxy-0.3.0" = "sha256-TEC51s/viqXUoEH9rJev8LdC2uHqefInNcarxeogePk=";
+    };
+  };
+
+  env = {
+    OPENSSL_NO_VENDOR = 1;
+  };
+
+  postPatch = ''
+    substituteInPlace ./tauri.conf.json \
+      --replace-fail '"distDir": "../dist",' '"distDir": "${ui}",' \
+      --replace-fail '"beforeBuildCommand": "pnpm run web:build"' '"beforeBuildCommand": ""'
+
+    # install mihomo
+    mkdir -p ./sidecar
+    ln -s ${mihomo}/bin/mihomo sidecar/verge-mihomo-${rust.envVars.rustHostPlatform}
+    ln -s ${mihomo-alpha}/bin/mihomo sidecar/verge-mihomo-alpha-${rust.envVars.rustHostPlatform}
```

Comment:
I think they are now called `$out/bin/verge-mihomo` and `$out/bin/verge-mihomo-alpha`.